README.mdt

```diff
@@ -0,0 +1,177 @@
+
+## Quick Start
+
+### 1. Register a JavaScript SDK Redirect URI
+
+The JavaScript SDK uses a special redirect URI to provide a simpler flow to retrieve authorization codes. The redirect uri takes the following form:
+
+```
+https://javascript-sdk.smartcar.com/redirect-${version}?app_origin=<Your Application's Origin>
+```
+
+The `app_origin` should be the location of where your website is located. The origin consists of just the protocol and host of your site without the resource path. Some example origins:
+
+#### Valid:
++ `https://example.com`
++ `https://myapp.example.com`
++ `http://localhost:8000`
+
+#### Invalid:
++ `https://example.com/some/path`
++ `http://localhost:8000/some/path`
++ `http://localhost:8000?foo=bar#baz`
+
+Once you have constructed your redirect uri, make sure to register it on the [Smartcar dashboard](https://dashboard.smartcar.com).
+
+### 2. Initialize Smartcar
+
+```javascript
+const smartcar = new Smartcar({
+  clientId: '<your-client-id>',
+  redirectUri: '<your-redirect-uri>',
+  scope: ['read_vehicle_info', 'read_odometer'],
+  onComplete: function(err, code) {
+    if (err) {
+      // handle errors from the authorization flow (i.e. user denies access)
+    }
+    // handle the returned code by sending it to your backend server
+    sendToBackend(code);
+  },
+});
+```
+
+**Reference:** [`new Smartcar(options)`](doc#new_Smartcar_new)
+
+> NOTE: See the full set of available scopes in the [Smartcar API Reference](https://smartcar.com/docs#get-all-vehicles) under "Required Permissions" for each endpoint.
+
+### 3. Launch the authorization flow
+
+Add a click handler to an HTML element:
+
+```javascript
+smartcar.addClickHandler({id: '#your-button-id'});
+```
+
+**Reference:** [`smartcar.addClickHandler(options)`](doc#Smartcar+addClickHandler)
+
+Alternatively, you can launch the authorization flow directly:
+
+```javascript
+smartcar.openDialog();
+```
+
+**Reference:** [`smartcar.openDialog(options)`](doc#Smartcar+openDialog)
+
+## Advanced
+
+The JavaScript SDK also can be used in alternative flows than the one described above. This section will cover some of its other usages.
+
+### Authorization URI Generation
+
+Normally the [`.addClickHandler()`](doc#Smartcar+addClickHandler) and [`.openDialog()`](doc#Smartcar+openDialog) methods are used to launch the authorization flow. However, if you would like to generate the OAuth authorization URL directly you can do so with the [`.getAuthUrl()`](doc#Smartcar+getAuthUrl) method.
+
+```javascript
+const url = smartcar.getAuthUrl();
+```
+
+**Reference:** [`smartcar.getAuthUrl(options)`](doc#Smartcar+getAuthUrl)
+
+### Server side redirect handling
+
+In a traditional OAuth implementation, the redirect uri is normally set to your application's server backend, rather than the JavaScript SDK redirect page used by this SDK. In this architecture you would recieve the authorization code on a backend route instead of the client-side `onComplete` callback.
```

Comment:
In a traditional OAuth implementation, the redirect **URI** is normally set to your application's server **back end**, rather than **to a** redirect page used by **an SDK**. In **the architecture of this JavaScript SDK**, you **receive** the authorization code on a **back-end** route instead of the client-side `onComplete` callback.

Reply:
I'm not sure whether I understand these two sentences correctly, so my edits might be wrong.